ui/panels-plugin/src/plugins/scatterplot/ScatterChartPanel.tsx

```diff
@@ -0,0 +1,165 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { PanelProps, useDataQueries } from '@perses-dev/plugin-system';
+import { Box, Skeleton } from '@mui/material';
+import { useMemo } from 'react';
+import { AbsoluteTimeRange, TraceValue } from '@perses-dev/core';
+import { EChartsOption, SeriesOption } from 'echarts';
+import { ErrorAlert } from '@perses-dev/components';
+import { getUnixTime } from 'date-fns';
+import { Scatterplot } from './Scatterplot';
+import { ScatterChartOptions } from './scatter-chart-model';
+
+export interface EChartTraceValue extends Omit<TraceValue, 'startTimeUnixMs'> {
+  startTime: Date;
+}
+
+export function getUnixTimeRange(timeRange: AbsoluteTimeRange) {
```

Comment:
Can't remember exactly, but we might have a similar util in @perses-dev/core

Reply:
I’ve removed this in the latest commit. This was a helper function used by greaterThanSevenDays().